src/Command/RoutesCommand.php

```diff
@@ -61,6 +61,12 @@ public function execute(Arguments $args, ConsoleIo $io): ?int
             }
 
             $output[] = $item;
+
+            // Count route templates
+            if (!array_key_exists($route->template, $duplicateRoutesCounter)) {
+                $duplicateRoutesCounter[$route->template] = 0;
+            }
+            $duplicateRoutesCounter[$route->template]++;
```

Comment:
Isn't counting duplicates based on just the template too simplistic? For e.g. two routes could have the same template but match different URLs based on regex for the route placeholders.

I am just concerned if showing routes which aren't actually duplicates as so would be more confusing to users.

Reply:
I used the same method which is used in the Debugkit here https://github.com/cakephp/debug_kit/blob/4.x/templates/element/routes_panel.php#L80

Will have to read more into route placeholders since I am not 100% sure how to check for that.

Reply:
But looking at my example above I myself have a plugin which uses the `/tools/{controller}` and `/tools/{controller}/{action}/*` which is also defined by the Tools Plugin of dereuromark.

I definitely don't have a problem here since my plugin doesn't have controllers which are also used inside dereuromark tool plugin but it could be a problem.

Maybe adjusting the text to something like "The following routes could have unexpected behaviors" or something like that is better?

Reply:
I for e.g. have multiple routes with the template `/p/{id}/{slug}` which map to different controllers since the regex for each `id` is different.

Reply:
Well then what would your solution for such a regex/pattern based route collision be?

Reply:
Personally I  would just skip this duplicates checking since any addionatal checks would be too complicated. But if others feels that this can be useful then the language for message above that list should be changed, stating they **could** be duplicate.

Reply:
Could this be an option? Show "duplicate" strings. Show duplicate regexes or whatever.

Reply:
That's a possibility but I think this feature mainly helps newbies and they won't know enough to use the flag.

Reply:
Hmm, in that case, I think different color coding for "possible duplicates" vs "duplicate" would help.

Reply:
I now adjusted it to `The following possible route collisions were detected`